trie.h

```diff
@@ -19,6 +19,7 @@ typedef struct
     const uint8_t   *completions;       // packed completions strings buffer
     uint8_t         completion_max_len; // max len of all completion strings
     uint8_t         max_backspaces;     // max backspaces for all completions
+    st_key_stack_t * const  key_stack;  // key stack used for searches
```

Comment:
const in the middle, a few lines below(line 50 in the same file) you use `const type *name;`

Reply:
They don't mean the same thing.

`st_key_stack_t * const` is a constant pointer to an `st_key_stack_t`. That means that the pointer always points to the same `st_key_stack_t`, but the values in the struct can be updated.

`const uint8_t *completions` is a pointer to a `const uint8_t`. You can change the pointer, but you can't change the value of the referenced `const uint8_t`.

Reply:
Matt is correct. Easy way to remember when reading const in C is: `const binds left (except if it's at the start)`.
We should also change the defs of other pointers so that they can't be changed, but I didn't want to touch more than I needed in this PR.